### PR TITLE
 fix bug for extended ASCII code

### PR DESCRIPTION
**Pull request by alivxxx**
When the last character is extended ASCII code, `LittleEndian::read_int` will extend the sign bit and result in the wrong result. This PR fixes this by replacing the `LittleEndian::read_int` with `LittleEndian::read_uint`.

### CODE REVIEW

**Comment by coocood** [comment]
LGTM

**Comment by alivxxx** [comment]
Following is the C code for remaining part of the string. From it, we can see that the remaining part is extended with 0, not the sign bit. For example, [0xFF] is the same as [0xFF, 0, 0, 0, 0, 0, 0, 0], not [0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF].
```
  uint64_t k1 = 0;
  uint64_t k2 = 0;

  switch(len & 15)
  {
  case 15: k2 ^= (uint64_t)(tail[14]) << 48;
  case 14: k2 ^= (uint64_t)(tail[13]) << 40;
  case 13: k2 ^= (uint64_t)(tail[12]) << 32;
  case 12: k2 ^= (uint64_t)(tail[11]) << 24;
  case 11: k2 ^= (uint64_t)(tail[10]) << 16;
  case 10: k2 ^= (uint64_t)(tail[ 9]) << 8;
  case  9: k2 ^= (uint64_t)(tail[ 8]) << 0;
           k2 *= c2; k2  = ROTL64(k2,33); k2 *= c1; h2 ^= k2;

  case  8: k1 ^= (uint64_t)(tail[ 7]) << 56;
  case  7: k1 ^= (uint64_t)(tail[ 6]) << 48;
  case  6: k1 ^= (uint64_t)(tail[ 5]) << 40;
  case  5: k1 ^= (uint64_t)(tail[ 4]) << 32;
  case  4: k1 ^= (uint64_t)(tail[ 3]) << 24;
  case  3: k1 ^= (uint64_t)(tail[ 2]) << 16;
  case  2: k1 ^= (uint64_t)(tail[ 1]) << 8;
  case  1: k1 ^= (uint64_t)(tail[ 0]) << 0;
           k1 *= c1; k1  = ROTL64(k1,31); k1 *= c2; h1 ^= k1;
  };
```

**Comment by breezewish** [comment]
LGTM

**Review by BusyJay — APPROVED** [review]

File: src/lib.rs [hunk 1]
```diff
@@ -126,21 +126,21 @@ pub fn murmur3_x86_128<T :Read>(source: &mut T, seed: u32, out: &mut [u8]) {
...
                         h1 ^= process_h1_k_x86(LittleEndian::read_u32(&buf[0..4]));
-                        h2 ^= process_h2_k_x86(LittleEndian::read_int(&buf[4..size], size-4) as u32);
+                        h2 ^= process_h2_k_x86(LittleEndian::read_uint(&buf[4..size], size-4) as u32);
                     }
                     4 => {
                         h1 ^= process_h1_k_x86(LittleEndian::read_u32(&buf));
                     }
                     1...3 => {
-                        h1 ^= process_h1_k_x86(LittleEndian::read_int(&buf, size) as u32);
+                        h1 ^= process_h1_k_x86(LittleEndian::read_uint(&buf, size) as u32);
```
Comment (BusyJay) [inline src/lib.rs:143]: So why `uint` instead of `u32`?

Comment (alivxxx) [reply src/lib.rs:143]: `read_u32` will read 4 bytes of `buf`, but the `buf` may contain dirty number from the previous round.

**Pull request merged**